woocommerce/woocommerce
# Add data store attribute terms

### All Submissions:






### Changes proposed in this Pull Request:

Allows nested resources in our CRUD data stores.  Also adds the product attribute terms data store.

`some/path/{parent_id}/morepath/{id}`

Closes #33439, closes #33440, #33441 .

### How to test the changes in this Pull Request:

1. In your console run `wp.data.select('wc/admin/products/attributes/terms').getProductAttributeTerms({parent_id: 3});` - replacing `3` with an actual product attribute ID in your store.
2. Run `wp.data.select('wc/admin/products/attributes/terms').getProductAttributeTerms({id: 5, parent_id: 3});` Replacing `3` and `5` with actual attribute IDs and term IDs respectively in your store. 
3. Attempt to update a term using an attribute and term ID- `wp.data.dispatch('wc/admin/products/attributes/terms').updateProductAttributeTerm({id: 5, parent_id: 3 }, {name: 'Hooray!'});`
4. Attempt to delete a term - `wp.data.dispatch('wc/admin/products/attributes/terms').deleteProductAttributeTerm({id: 5, parent_id: 3 });`
5. Note that skipping the `parent_id` on any of these actions/resolvers stores an error in Redux - `wp.data.dispatch('wc/admin/products/attributes/terms').deleteProductAttributeTerm({id: 161 });

### Other information:




### FOR PR REVIEWER ONLY:
